core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala

```diff
@@ -77,6 +77,30 @@ class AppStatusStoreSuite extends SparkFunSuite {
     assert(store.count(classOf[CachedQuantile]) === 2)
   }
 
+  test("only successfull task have taskSummary") {
+    val store = new InMemoryStore()
+    (0 until 5).foreach { i => store.write(failedTaskData(i)) }
+    val appStore = new AppStatusStore(store).taskSummary(stageId, attemptId, uiQuantiles)
+    assert(appStore.size === 0)
+  }
+
+  test("summary should contain task metrics of only successfull tasks") {
+    val store = new InMemoryStore()
+    (0 until 5).foreach { i =>
+      if (i % 2 == 1) store.write(failedTaskData(i))
```

Comment:
Braces for if-else. I also personally think a for loop is more obvious here: `for (i <- 0 until 5) {`

Reply:
Done